yarn-plugin/sources/utils.ts

```diff
@@ -80,13 +159,37 @@ export const getSortedPackages = async (
   return packages;
 };
 
-export interface ManifestWithLicenseInfo {
-  name: string;
-  license?: ManifestLicenseValue;
-  licenses?: ManifestLicenseValue | ManifestLicenseValue[];
-  repository?: { url: string } | string;
-  homepage?: string;
-  author?: { name: string; url: string };
+export async function getPackageInfos(
+  packageManifest: ManifestWithLicenseInfo,
+): Promise<string> {
+  const { repository } = packageManifest;
+  const formattedRepository: string = getDownloadLocation(repository);
+  return formattedRepository;
 }
 
-type ManifestLicenseValue = string | { type: string };
+export function getDownloadLocation(
+  repository: { url: string } | string | undefined,
+): string {
+  if (
+    repository &&
+    typeof repository === "object" &&
+    isValidDownloadLocation(repository.url)
+  ) {
+    return repository.url;
+  } else {
+    return spdxNoAssertion;
+  }
+}
+
+function isValidDownloadLocation(downloadLocation: string): boolean {
+  return (
+    new RegExp(urlRegex).test(downloadLocation) &&
```

Comment:
Wouldn't it be more performant to run `new RegExp(urlRegex)` once and to store that in a constant. You than can run `.test(..)` for it multiple times.